external-crates/move/crates/move-compiler/src/expansion/translate.rs

```diff
@@ -3110,6 +3363,225 @@ fn exp_dotted(context: &mut Context, pdotted: Box<P::Exp>) -> Option<Box<E::ExpD
+                            P::Ellipsis::Ellipsis(eloc) => {
+                                ellipsis_locs.push(eloc);
+                            }
+                        }
+                    }
+                    let fields =
+                        named_fields(context, loc, "pattern", "sub-pattern", stripped_fields);
+                    check_ellipsis_usage(context, &ellipsis_locs);
+                    let ellipsis = ellipsis_locs.first().copied();
+                    sp(
+                        loc,
+                        EP::FieldConstructor(head_ctor_name, tys, fields, ellipsis),
+                    )
+                }
+                _ => error_pattern!(),
+            }
+        }
+        PP::Name(mut_, name_chain, pts_opt) => {
+            let head_ctor_name = context
+                .name_access_chain_to_module_access(Access::Variant, name_chain)
+                .and_then(|name| head_ctor_okay(context, name, true));
+            let tys = optional_types(context, pts_opt);
+            match head_ctor_name {
+                Some(sp!(loc, EM::Name(name))) => {
+                    let name_value = name.value;
+                    if !valid_local_variable_name(name_value) {
+                        let msg = format!(
+                            "Invalid pattern variable name '{}'. Pattern variable names must start \
+                            with 'a'..'z' (or '_')",
+                            name_value,
+                        );
+                        context
+                            .env()
+                            .add_diag(diag!(Declarations::InvalidName, (name.loc, msg)));
+                        error_pattern!()
+                    } else {
+                        sp(loc, EP::Binder(mut_, Var(name)))
+                    }
+                }
+                Some(head_ctor_name @ sp!(_, EM::Variant(_, _))) => {
+                    if let Some(mloc) = mut_ {
+                        let msg = "'mut' can only be used with variable bindings in patterns";
+                        let nmsg = "This refers to a variant, not a variable binding";
+                        context.env().add_diag(diag!(
+                            Declarations::InvalidName,
+                            (mloc, msg),
+                            (head_ctor_name.loc, nmsg)
+                        ));
+                        error_pattern!()
+                    } else {
+                        sp(loc, EP::HeadConstructor(head_ctor_name, tys))
+                    }
+                }
+                _ => error_pattern!(),
+            }
+        }
+        PP::Literal(v) => {
+            if let Some(v) = value(&mut context.defn_context, v) {
+                sp(loc, EP::Literal(v))
+            } else {
+                assert!(context.env().has_errors());
+                error_pattern!()
+            }
+        }
+        PP::Or(lhs, rhs) => sp(
+            loc,
+            EP::Or(
```

Comment:
I think this is fine (but maybe should be a warning?). I could see folks doing this as part of development cycle/debugging stuff possibly.